File: eXoGadget/server/php/src/socialdata/opensocial/model/Person.php
<?php
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 */

/**
 * see
 * http://code.google.com/apis/opensocial/docs/0.7/reference/opensocial.Person.Field.html
 *
 */
class Person {
	public $aboutMe;
	public $activities;
	public $addresses;
	public $age;
	public $bodyType;
	public $books;
	public $cars;
	public $children;
	public $currentLocation;
	public $dateOfBirth;
	public $drinker;
	public $emails;
	public $ethnicity;
	public $fashion;
	public $food;
	public $gender;
	public $happiestWhen;
	public $heroes;
	public $humor;
	public $id;
	public $interests;
	public $jobInterests;
	public $jobs;
	public $languagesSpoken;
	public $livingArrangement;
	public $lookingFor;
	public $movies;
	public $music;
	public $name;
	public $nickname;
	public $pets;
	public $phoneNumbers;
	public $politicalViews;
	public $profileSong;
	public $profileUrl;
	public $profileVideo;
	public $quotes;
	public $relationshipStatus;
	public $religion;
	public $romance;
	public $scaredOf;
	public $schools;
	public $sexualOrientation;
	public $smoker;
	public $sports;
	public $status;
	public $tags;
	public $thumbnailUrl;
	public $timeZone;
	public $turnOffs;
	public $turnOns;
	public $tvShows;
	public $urls;
	
	// Note: Not in the opensocial js person object directly
	public $isOwner = false;
	public $isViewer = false;
	
	public function __construct($id, $name)
	{
		$this->id = $id;
		$this->name = $name;
	}
	
	public function getAboutMe()
	{
		return $this->aboutMe;
	}
	
	public function setAboutMe($aboutMe)
	{
		$this->aboutMe = $aboutMe;
	}
	
	public function getActivities()
	{
		return $this->activities;
	}
	
	public function setActivities($activities)
	{
		$this->activities = $activities;
	}
	
	public function getAddresses()
	{
		return $this->addresses;
	}
	
	public function setAddresses($addresses)
	{
		$this->addresses = $addresses;
	}
	
	public function getAge()
	{
		return $this->age;
	}
	
	public function setAge($age)
	{
		$this->age = $age;
	}
	
	public function getBodyType()
	{
		return $this->bodyType;
	}
	
	public function setBodyType($bodyType)
	{
		$this->bodyType = $bodyType;
	}
	
	public function getBooks()
	{
		return $this->books;
	}
	
	public function setBooks($books)
	{
		$this->books = $books;
	}
	
	public function getCars()
	{
		return $this->cars;
	}
	
	public function setCars($cars)
	{
		$this->cars = $cars;
	}
	
	public function getChildren()
	{
		return $this->children;
	}
	
	public function setChildren($children)
	{
		$this->children = $children;
	}
	
	public function getCurrentLocation()
	{
		return $this->currentLocation;
	}
	
	public function setCurrentLocation($currentLocation)
	{
		$this->currentLocation = $currentLocation;
	}
	
	public function getDateOfBirth()
	{
		return $this->dateOfBirth;
	}
	
	public function setDateOfBirth($dateOfBirth)
	{
		$this->dateOfBirth = $dateOfBirth;
	}
	
	public function getDrinker()
	{
		return $this->$this->drinker;
	}
	
	public function setDrinker($newDrinker)
	{
		$this->drinker = $newDrinker;
	}
	
	public function getEmails()
	{
		return $this->emails;
	}
	
	public function setEmails($emails)
	{
		$this->emails = $emails;
	}
	
	public function getEthnicity()
	{
		return $this->ethnicity;
	}
	
	public function setEthnicity($ethnicity)
	{
		$this->ethnicity = $ethnicity;
	}
	
	public function getFashion()
	{
		return $this->fashion;
	}
	
	public function setFashion($fashion)
	{
		$this->fashion = $fashion;
	}
	
	public function getFood()
	{
		return $this->food;
	}
	
	public function setFood($food)
	{
		$this->food = $food;
	}
	
	public function getGender()
	{
		return $this->$this->gender;
	}
	
	public function setGender($newGender)
	{
		$this->gender = $newGender;
	}
	
	public function getHappiestWhen()
	{
		return $this->happiestWhen;
	}
	
	public function setHappiestWhen($happiestWhen)
	{
		$this->happiestWhen = $happiestWhen;
	}
	
	public function getHeroes()
	{
		return $this->heroes;
	}
	
	public function setHeroes($heroes)
	{
		$this->heroes = $heroes;
	}
	
	public function getHumor()
	{
		return $this->humor;
	}
	
	public function setHumor($humor)
	{
		$this->humor = $humor;
	}
	
	public function getId()
	{
		return $this->id;
	}
	
	public function setId($id)
	{
		$this->id = $id;
	}
	
	public function getInterests()
	{
		return $this->interests;
	}
	
	public function setInterests($interests)
	{
		$this->interests = $interests;
	}
	
	public function getJobInterests()
	{
		return $this->jobInterests;
	}
	
	public function setJobInterests($jobInterests)
	{
		$this->jobInterests = $jobInterests;
	}
	
	public function getJobs()
	{
		return $this->jobs;
	}
	
	public function setJobs($jobs)
	{
		$this->jobs = $jobs;
	}
	
	public function getLanguagesSpoken()
	{
		return $this->languagesSpoken;
	}
	
	public function setLanguagesSpoken($languagesSpoken)
	{
		$this->languagesSpoken = $languagesSpoken;
	}
	
	public function getLivingArrangement()
	{
		return $this->livingArrangement;
	}
	
	public function setLivingArrangement($livingArrangement)
	{
		$this->livingArrangement = $livingArrangement;
	}
	
	public function getLookingFor()
	{
		return $this->lookingFor;
	}
	
	public function setLookingFor($lookingFor)
	{
		$this->lookingFor = $lookingFor;
	}
	
	public function getMovies()
	{
		return $this->movies;
	}
	
	public function setMovies($movies)
	{
		$this->movies = $movies;
	}
	
	public function getMusic()
	{
		return $this->music;
	}
	
	public function setMusic($music)
	{
		$this->music = $music;
	}
	
	public function getName()
	{
		return $this->name;
	}
	
	public function setName($name)
	{
		$this->name = $name;
	}
	
	public function getNickname()
	{
		return $this->nickname;
	}
	
	public function setNickname($nickname)
	{
		$this->nickname = $nickname;
	}
	
	public function getPets()
	{
		return $this->pets;
	}
	
	public function setPets($pets)
	{
		$this->pets = $pets;
	}
	
	public function getPhoneNumbers()
	{
		return $this->phoneNumbers;
	}
	
	public function setPhoneNumbers($phoneNumbers)
	{
		$this->phoneNumbers = $phoneNumbers;
	}
	
	public function getPoliticalViews()
	{
		return $this->politicalViews;
	}
	
	public function setPoliticalViews($politicalViews)
	{
		$this->politicalViews = $politicalViews;
	}
	
	public function getProfileSong()
	{
		return $this->profileSong;
	}
	
	public function setProfileSong($profileSong)
	{
		$this->profileSong = $profileSong;
	}
	
	public function getProfileUrl()
	{
		return $this->profileUrl;
	}
	
	public function setProfileUrl($profileUrl)
	{
		$this->profileUrl = $profileUrl;
	}
	
	public function getProfileVideo()
	{
		return $this->profileVideo;
	}
	
	public function setProfileVideo($profileVideo)
	{
		$this->profileVideo = $profileVideo;
	}
	
	public function getQuotes()
	{
		return $this->quotes;
	}
	
	public function setQuotes($quotes)
	{
		$this->quotes = $quotes;
	}
	
	public function getRelationshipStatus()
	{
		return $this->relationshipStatus;
	}
	
	public function setRelationshipStatus($relationshipStatus)
	{
		$this->relationshipStatus = $relationshipStatus;
	}
	
	public function getReligion()
	{
		return $this->religion;
	}
	
	public function setReligion($religion)
	{
		$this->religion = $religion;
	}
	
	public function getRomance()
	{
		return $this->romance;
	}
	
	public function setRomance($romance)
	{
		$this->romance = $romance;
	}
	
	public function getScaredOf()
	{
		return $this->scaredOf;
	}
	
	public function setScaredOf($scaredOf)
	{
		$this->scaredOf = $scaredOf;
	}
	
	public function getSchools()
	{
		return $this->schools;
	}
	
	public function setSchools($schools)
	{
		$this->schools = $schools;
	}
	
	public function getSexualOrientation()
	{
		return $this->sexualOrientation;
	}
	
	public function setSexualOrientation($sexualOrientation)
	{
		$this->sexualOrientation = $sexualOrientation;
	}
	
	public function getSmoker()
	{
		return $this->$this->smoker;
	}
	
	public function setSmoker($newSmoker)
	{
		$this->smoker = $newSmoker;
	}
	
	public function getSports()
	{
		return $this->sports;
	}
	
	public function setSports($sports)
	{
		$this->sports = $sports;
	}
	
	public function getStatus()
	{
		return $this->status;
	}
	
	public function setStatus($status)
	{
		$this->status = $status;
	}
	
	public function getTags()
	{
		return $this->tags;
	}
	
	public function setTags($tags)
	{
		$this->tags = $tags;
	}
	
	public function getThumbnailUrl()
	{
		return $this->thumbnailUrl;
	}
	
	public function setThumbnailUrl($thumbnailUrl)
	{
		$this->thumbnailUrl = $thumbnailUrl;
	}
	
	public function getTimeZone()
	{
		return $this->timeZone;
	}
	
	public function setTimeZone($timeZone)
	{
		$this->timeZone = $timeZone;
	}
	
	public function getTurnOffs()
	{
		return $this->turnOffs;
	}
	
	public function setTurnOffs($turnOffs)
	{
		$this->turnOffs = $turnOffs;
	}
	
	public function getTurnOns()
	{
		return $this->turnOns;
	}
	
	public function setTurnOns($turnOns)
	{
		$this->turnOns = $turnOns;
	}
	
	public function getTvShows()
	{
		return $this->tvShows;
	}
	
	public function setTvShows($tvShows)
	{
		$this->tvShows = $tvShows;
	}
	
	public function getUrls()
	{
		return $this->urls;
	}
	
	public function setUrls($urls)
	{
		$this->urls = $urls;
	}
	
	public function getIsOwner()
	{
		return $this->isOwner;
	}
	
	public function setIsOwner($isOwner)
	{
		$this->isOwner = $isOwner;
	}
	
	public function getIsViewer()
	{
		return $this->isViewer;
	}
	
	public function setIsViewer($isViewer)
	{
		$this->isViewer = $isViewer;
	}
}

File: gadgets/shindig/java/social-api/src/main/java/org/apache/shindig/social/opensocial/model/Person.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 */
package org.apache.shindig.social.opensocial.model;

import org.apache.shindig.social.AbstractGadgetData;
import org.apache.shindig.social.Mandatory;

import java.util.Date;
import java.util.List;

/**
 * see
 * http://code.google.com/apis/opensocial/docs/0.7/reference/opensocial.Person.Field.html
 *
 */
public class Person extends AbstractGadgetData {

  public static enum Field {
    ABOUT_ME("aboutMe"),
    ACTIVITIES("activities"),
    ADDRESSES("addresses"),
    AGE("age"),
    BODY_TYPE("bodyType"),
    BOOKS("books"),
    CARS("cars"),
    CHILDREN("children"),
    CURRENT_LOCATION("currentLocation"),
    DATE_OF_BIRTH("dateOfBirth"),
    DRINKER("drinker"),
    EMAILS("emails"),
    ETHNICITY("ethnicity"),
    FASHION("fashion"),
    FOOD("food"),
    GENDER("gender"),
    HAPPIEST_WHEN("happiestWhen"),
    HEROES("heroes"),
    HUMOR("humor"),
    ID("id"),
    INTERESTS("interests"),
    JOB_INTERESTS("jobInterests"),
    JOBS("jobs"),
    LANGUAGES_SPOKEN("languagesSpoken"),
    LAST_UPDATED("updated"),
    LIVING_ARRANGEMENT("livingArrangement"),
    LOOKING_FOR("lookingFor"),
    MOVIES("movies"),
    MUSIC("music"),
    NAME("name"),
    NICKNAME("nickname"),
    PETS("pets"),
    PHONE_NUMBERS("phoneNumbers"),
    POLITICAL_VIEWS("politicalViews"),
    PROFILE_SONG("profileSong"),
    PROFILE_URL("profileUrl"),
    PROFILE_VIDEO("profileVideo"),
    QUOTES("quotes"),
    RELATIONSHIP_STATUS("relationshipStatus"),
    RELIGION("religion"),
    ROMANCE("romance"),
    SCARED_OF("scaredOf"),
    SCHOOLS("schools"),
    SEXUAL_ORIENTATION("sexualOrientation"),
    SMOKER("smoker"),
    SPORTS("sports"),
    STATUS("status"),
    TAGS("tags"),
    THUMBNAIL_URL("thumbnailUrl"),
    TIME_ZONE("timeZone"),
    TURN_OFFS("turnOffs"),
    TURN_ONS("turnOns"),
    TV_SHOWS("tvShows"),
    URLS("urls");

    private final String jsonString;

    private Field(String jsonString) {
      this.jsonString = jsonString;
    }

    @Override
    public String toString() {
      return this.jsonString;
    }
  }

  private String aboutMe;
  private List<String> activities;
  private List<Address> addresses;
  private Integer age;
  private BodyType bodyType;
  private List<String> books;
  private List<String> cars;
  private String children;
  private Address currentLocation;
  private Date dateOfBirth;
  private Enum<Enum.Drinker> drinker;
  private List<Email> emails;
  private String ethnicity;
  private String fashion;
  private List<String> food;
  private Enum<Enum.Gender> gender;
  private String happiestWhen;
  private List<String> heroes;
  private String humor;
  private String id;
  private List<String> interests;
  private String jobInterests;
  private List<Organization> jobs;
  private List<String> languagesSpoken;
  private Date updated;
  private String livingArrangement;
  private String lookingFor;
  private List<String> movies;
  private List<String> music;
  private Name name;
  private String nickname;
  private String pets;
  private List<Phone> phoneNumbers;
  private String politicalViews;
  private Url profileSong;
  private String profileUrl;
  private Url profileVideo;
  private List<String> quotes;
  private String relationshipStatus;
  private String religion;
  private String romance;
  private String scaredOf;
  private List<Organization> schools;
  private String sexualOrientation;
  private Enum<Enum.Smoker> smoker;
  private List<String> sports;
  private String status;
  private List<String> tags;
  private String thumbnailUrl;
  private Long timeZone;
  private List<String> turnOffs;
  private List<String> turnOns;
  private List<String> tvShows;
  private List<Url> urls;

  // Note: Not in the opensocial js person object directly
  private boolean isOwner = false;
  private boolean isViewer = false;

  public Person(String id, Name name) {
    this.id = id;
    this.name = name;
  }

  public String getAboutMe() {
    return aboutMe;
  }

  public void setAboutMe(String aboutMe) {
    this.aboutMe = aboutMe;
  }

  public List<String> getActivities() {
    return activities;
  }

  public void setActivities(List<String> activities) {
    this.activities = activities;
  }

  public List<Address> getAddresses() {
    return addresses;
  }

  public void setAddresses(List<Address> addresses) {
    this.addresses = addresses;
  }

  public Integer getAge() {
    return age;
  }

  public void setAge(Integer age) {
    this.age = age;
  }

  public BodyType getBodyType() {
    return bodyType;
  }

  public void setBodyType(BodyType bodyType) {
    this.bodyType = bodyType;
  }

  public List<String> getBooks() {
    return books;
  }

  public void setBooks(List<String> books) {
    this.books = books;
  }

  public List<String> getCars() {
    return cars;
  }

  public void setCars(List<String> cars) {
    this.cars = cars;
  }

  public String getChildren() {
    return children;
  }

  public void setChildren(String children) {
    this.children = children;
  }

  public Address getCurrentLocation() {
    return currentLocation;
  }

  public void setCurrentLocation(Address currentLocation) {
    this.currentLocation = currentLocation;
  }

  public Date getDateOfBirth() {
    return dateOfBirth;
  }

  public void setDateOfBirth(Date dateOfBirth) {
    this.dateOfBirth = dateOfBirth;
  }

  public Enum<Enum.Drinker> getDrinker() {
    return this.drinker;
  }

  public void setDrinker(Enum<Enum.Drinker> newDrinker) {
    this.drinker = newDrinker;
  }

  public List<Email> getEmails() {
    return emails;
  }

  public void setEmails(List<Email> emails) {
    this.emails = emails;
  }

  public String getEthnicity() {
    return ethnicity;
  }

  public void setEthnicity(String ethnicity) {
    this.ethnicity = ethnicity;
  }

  public String getFashion() {
    return fashion;
  }

  public void setFashion(String fashion) {
    this.fashion = fashion;
  }

  public List<String> getFood() {
    return food;
  }

  public void setFood(List<String> food) {
    this.food = food;
  }

  public Enum<Enum.Gender> getGender() {
    return this.gender;
  }

  public void setGender(Enum<Enum.Gender> newGender) {
    this.gender = newGender;
  }

  public String getHappiestWhen() {
    return happiestWhen;
  }

  public void setHappiestWhen(String happiestWhen) {
    this.happiestWhen = happiestWhen;
  }

  public List<String> getHeroes() {
    return heroes;
  }

  public void setHeroes(List<String> heroes) {
    this.heroes = heroes;
  }

  public String getHumor() {
    return humor;
  }

  public void setHumor(String humor) {
    this.humor = humor;
  }

  @Mandatory
  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public List<String> getInterests() {
    return interests;
  }

  public void setInterests(List<String> interests) {
    this.interests = interests;
  }

  public String getJobInterests() {
    return jobInterests;
  }

  public void setJobInterests(String jobInterests) {
    this.jobInterests = jobInterests;
  }

  public List<Organization> getJobs() {
    return jobs;
  }

  public void setJobs(List<Organization> jobs) {
    this.jobs = jobs;
  }

  public List<String> getLanguagesSpoken() {
    return languagesSpoken;
  }

  public void setLanguagesSpoken(List<String> languagesSpoken) {
    this.languagesSpoken = languagesSpoken;
  }

  public Date getUpdated() {
    return updated;
  }

  public void setUpdated(Date updated) {
    this.updated = updated;
  }

  public String getLivingArrangement() {
    return livingArrangement;
  }

  public void setLivingArrangement(String livingArrangement) {
    this.livingArrangement = livingArrangement;
  }

  public String getLookingFor() {
    return lookingFor;
  }

  public void setLookingFor(String lookingFor) {
    this.lookingFor = lookingFor;
  }

  public List<String> getMovies() {
    return movies;
  }

  public void setMovies(List<String> movies) {
    this.movies = movies;
  }

  public List<String> getMusic() {
    return music;
  }

  public void setMusic(List<String> music) {
    this.music = music;
  }

  @Mandatory
  public Name getName() {
    return name;
  }

  public void setName(Name name) {
    this.name = name;
  }

  public String getNickname() {
    return nickname;
  }

  public void setNickname(String nickname) {
    this.nickname = nickname;
  }

  public String getPets() {
    return pets;
  }

  public void setPets(String pets) {
    this.pets = pets;
  }

  public List<Phone> getPhoneNumbers() {
    return phoneNumbers;
  }

  public void setPhoneNumbers(List<Phone> phoneNumbers) {
    this.phoneNumbers = phoneNumbers;
  }

  public String getPoliticalViews() {
    return politicalViews;
  }

  public void setPoliticalViews(String politicalViews) {
    this.politicalViews = politicalViews;
  }

  public Url getProfileSong() {
    return profileSong;
  }

  public void setProfileSong(Url profileSong) {
    this.profileSong = profileSong;
  }

  public String getProfileUrl() {
    return profileUrl;
  }

  public void setProfileUrl(String profileUrl) {
    this.profileUrl = profileUrl;
  }

  public Url getProfileVideo() {
    return profileVideo;
  }

  public void setProfileVideo(Url profileVideo) {
    this.profileVideo = profileVideo;
  }

  public List<String> getQuotes() {
    return quotes;
  }

  public void setQuotes(List<String> quotes) {
    this.quotes = quotes;
  }

  public String getRelationshipStatus() {
    return relationshipStatus;
  }

  public void setRelationshipStatus(String relationshipStatus) {
    this.relationshipStatus = relationshipStatus;
  }

  public String getReligion() {
    return religion;
  }

  public void setReligion(String religion) {
    this.religion = religion;
  }

  public String getRomance() {
    return romance;
  }

  public void setRomance(String romance) {
    this.romance = romance;
  }

  public String getScaredOf() {
    return scaredOf;
  }

  public void setScaredOf(String scaredOf) {
    this.scaredOf = scaredOf;
  }

  public List<Organization> getSchools() {
    return schools;
  }

  public void setSchools(List<Organization> schools) {
    this.schools = schools;
  }

  public String getSexualOrientation() {
    return sexualOrientation;
  }

  public void setSexualOrientation(String sexualOrientation) {
    this.sexualOrientation = sexualOrientation;
  }

  public Enum<Enum.Smoker> getSmoker() {
    return this.smoker;
  }

  public void setSmoker(Enum<Enum.Smoker> newSmoker) {
    this.smoker = newSmoker;
  }

  public List<String> getSports() {
    return sports;
  }

  public void setSports(List<String> sports) {
    this.sports = sports;
  }

  public String getStatus() {
    return status;
  }

  public void setStatus(String status) {
    this.status = status;
  }

  public List<String> getTags() {
    return tags;
  }

  public void setTags(List<String> tags) {
    this.tags = tags;
  }

  public String getThumbnailUrl() {
    return thumbnailUrl;
  }

  public void setThumbnailUrl(String thumbnailUrl) {
    this.thumbnailUrl = thumbnailUrl;
  }

  public Long getTimeZone() {
    return timeZone;
  }

  public void setTimeZone(Long timeZone) {
    this.timeZone = timeZone;
  }

  public List<String> getTurnOffs() {
    return turnOffs;
  }

  public void setTurnOffs(List<String> turnOffs) {
    this.turnOffs = turnOffs;
  }

  public List<String> getTurnOns() {
    return turnOns;
  }

  public void setTurnOns(List<String> turnOns) {
    this.turnOns = turnOns;
  }

  public List<String> getTvShows() {
    return tvShows;
  }

  public void setTvShows(List<String> tvShows) {
    this.tvShows = tvShows;
  }

  public List<Url> getUrls() {
    return urls;
  }

  public void setUrls(List<Url> urls) {
    this.urls = urls;
  }

  public boolean getIsOwner() {
    return isOwner;
  }

  public void setIsOwner(boolean isOwner) {
    this.isOwner = isOwner;
  }

  public boolean getIsViewer() {
    return isViewer;
  }

  public void setIsViewer(boolean isViewer) {
    this.isViewer = isViewer;
  }
}
